Source/Core/DolphinQt/Achievements/AchievementBox.cpp

```diff
@@ -87,6 +90,16 @@ void AchievementBox::UpdateData()
     m_status->setText(tr("Locked"));
   }
 
+  UpdateProgress();
```

Comment:
I think this is safe because you're using a `recursive_mutex` but you could consider putting the update data in its own scope.  Not sure if that would be more optimal or not..